observ/watcher.py

```diff
@@ -73,7 +74,37 @@ def run(self) -> None:
             old_value = self.value
             self.value = value
             if self.callback:
-                self.callback(old_value, self.value)
+                self.run_callback(old_value, self.value)
+
+    def run_callback(self, old, new):
+        """
+        Runs the callback. When the number of arguments is still unknown
+        for the callback, it will fall into the try/except contstruct
+        to figure out the right number of arguments.
+        After running the callback one time, the number of arguments
+        is known and the callback can be called with the correct
+        amount of arguments.
+        """
+        if self._number_of_callback_args == 1:
+            self.callback(new)
+            return
+        if self._number_of_callback_args == 2:
+            self.callback(old, new)
```

Comment:
I feel like maybe new, old, makes more sense

Reply:
Yeah, let's swap it, had some doubts on whether it was worth the effort but now would be a good time.